src/test/java/org/beanpod/switchboard/controller/DeviceControllerTest.java

```diff
@@ -126,23 +124,23 @@ final void testDeleteDevice(){
     //When a device is unavailable in the DB
     @Test
     final void testDeleteDeviceNotExisting(){
-        assertThrows(DeviceNotFoundException.class, () -> {
+        assertThrows(ExceptionType.DeviceNotFoundException.class, () -> {
             deviceController.deleteDevice("Not Available device");
         }, "DeviceNotFoundException should have been thrown.");
     }
 
     //When a device is available in the DB
     @Test
     final void testUpdateDevice(){
-        DeviceDTO deviceDTO1 = deviceMapper.toDeviceDTO(device1);
 
         when(deviceService.findDevice("1")).thenReturn(java.util.Optional.of(device1));
-        when(deviceService.updateDevice("1",deviceDTO1)).thenReturn(Integer.valueOf(1));
-
-        ResponseEntity<String> response = deviceController.updateDevice("1", deviceDTO1);
+        device1.setStatus("Stopped");
+        when(deviceService.save(device1)).thenReturn(device1);
+        DeviceDTO deviceDTO1 = deviceMapper.toDeviceDTO(device1);
```

Comment:
I'd recommend adding a `DeviceDTO` fixture in `DeviceFixture` instead of mapping it.

Reply:
Mapping it increases test coverage

Reply:
Maybe we should have separate tests that check just the mapper. but since it's a really simple mapper, that might be overkill

Reply:
its a mapper that is auto generated from mapstruct, i dont know if its a good idea to test manually.

Reply:
I'm a little late to the party,  but from my point of view the mappers are a library and coverage of those should be done during integration tests.